api/handlers.go

```diff
@@ -742,6 +917,47 @@ func (si *ServerImplementation) fetchAssetBalances(ctx context.Context, options
 	return balances, round, nil
 }
 
+// fetchAssetHoldings fetches all balances from a query and converts them into
+// generated.AssetHolding objects
+func (si *ServerImplementation) fetchAssetHoldings(ctx context.Context, options idb.AssetBalanceQuery) ([]generated.AssetHolding, uint64 /*round*/, error) {
+	var round uint64
+	balances := make([]generated.AssetHolding, 0)
+	err := callWithTimeout(ctx, si.log, si.timeout, func(ctx context.Context) error {
+		var assetbalchan <-chan idb.AssetBalanceRow
+		assetbalchan, round = si.db.AssetBalances(ctx, options)
+
+		for row := range assetbalchan {
+			if row.Error != nil {
+				return row.Error
+			}
+
+			addr := basics.Address{}
+			if len(row.Address) != len(addr) {
```

Comment:
I started trying this, from changing `decodeAddress` to return a basics.Address, but then abandoned it because I didn't want to change things a lot of different places. I think I will leave this to someone in the future to make an across-the-board change rather than have different checks/byte-to-address conversions take place in different places through the handles and DB code.